python/cudf/cudf/core/buffer.py

```diff
@@ -112,7 +118,7 @@ def serialize(self):
         return header, frames
 
     @classmethod
-    def deserialize(cls, header, frames):
+    def deserialize(cls, header: dict, frames: list) -> "Buffer":
```

Comment:
warning: series of annoying questions.

- Is `Buffer` in quotes because it's not in scope?
- Should it be in scope?
- What happens if we change the name of `Buffer` without updating the quote?

Reply:
This is explained here:

https://github.com/python/mypy/issues/3661

We can also use `from __future__ import annotations` and forgo the quotes if that's preferable?

Reply:
I'd be a +1 to doing that

Reply:
Done - I think I caught all instances